### PR TITLE
Fix setting of metrics compression type in RealtimeSegmentConverter

### PR DESCRIPTION
**Pull request by harold-kfuse**
## Description
Introduce a method in SegmentGeneratorConfig that allows
setting a specific column's compression type.
Use this method instead when setting metric compression type.
Otherwise, no dictionary config compression type for other columns
are getting reset to default.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/RealtimeSegmentConverter.java [hunk 1]
```diff
@@ -94,14 +93,14 @@ public void build(@Nullable SegmentVersion segmentVersion, ServerMetrics serverM
...
-      Map<String, ChunkCompressionType> columnToCompressionType = new HashMap<>();
       for (String column : _noDictionaryColumns) {
         FieldSpec fieldSpec = _dataSchema.getFieldSpecFor(column);
+        // Compression type is already configured in the constructor of genConfig.
+        // Only update metrics compression type.
         if (fieldSpec.getFieldType().equals(FieldSpec.FieldType.METRIC)) {
-          columnToCompressionType.put(column, ChunkCompressionType.PASS_THROUGH);
+          genConfig.setRawIndexCompressionType(column, ChunkCompressionType.PASS_THROUGH);
         }
       }
```
Comment (Jackie-Jiang) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/RealtimeSegmentConverter.java:103]: I think this part of the logic can be removed completely. In `SegmentColumnarIndexCreator.getColumnCompressionType()`, the compression type for metrics are default to `PASS_THROUGH`

**Review by Jackie-Jiang — COMMENTED** [review]
Good catch

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/RealtimeSegmentConverter.java (hunk 1 above)
Comment (harold-kfuse) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/RealtimeSegmentConverter.java:103]: Thanks. You mean I can delete this whole block? (line 96-103)?

Comment (Jackie-Jiang) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/RealtimeSegmentConverter.java:103]: Yes. We should not blindly override all metric compression to PASS_THROUGH even if it is explicitly configured differently

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**